src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1029,11 +1029,10 @@ private function addValidationSection(ArrayNodeDefinition $rootNode, callable $e
                     if (isset($v['enable_annotations'])) {
                         trigger_deprecation('symfony/framework-bundle', '6.4', 'Option "enable_annotations" at "framework.validation" is deprecated. Use the "enable_attributes" option instead.');
 
-                        if (!isset($v['enable_attributes'])) {
-                            $v['enable_attributes'] = $v['enable_annotations'];
-                        } else {
+                        if (isset($v['enable_attributes'])) {
```

Comment:
even this could be a BC break if it is done in `->validate()` as this would break if one file uses the new setting and another uses the old one.
Rewriting from the old one to the new one should be done in `beforeNormalization`